Mathlib/Analysis/NormedSpace/Connected.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2023 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.Convex.Topology
+import Mathlib.LinearAlgebra.Dimension
+import Mathlib.Topology.Algebra.Module.Cardinality
+
+/-!
+# Connectedness of subsets of vector spaces
+
+We show several results related to the (path)-connectedness of subsets of real vector spaces:
+* `Set.Countable.isPathConnected_compl_of_one_lt_rank` asserts that the complement of a countable
+  set is path-connected in a space of dimension `> 1`.
+* `isPathConnected_compl_singleton_of_one_lt_rank` is the special case of the complement of a
+  singleton.
+* `isPathConnected_sphere` shows that any sphere is path-connected in dimension `> 1`.
+
+Statements with connectedness instead of path-connectedness are also given.
+-/
+
+open Convex Set Metric
+
+section TopologicalVectorSpace
+
+variable {E : Type*} [AddCommGroup E] [Module ℝ E]
+
+lemma segment_inter_eq_endpoint_of_linearIndependent
+    {x y : E} (h : LinearIndependent ℝ ![x, y]) {s t : ℝ} (hs : s ≠ t) (c : E) :
+    [c + x -[ℝ] c + t • y] ∩ [c + x -[ℝ] c + s • y] ⊆ {c + x} := by
```

Comment:
I've added more API for linear independence of pairs and refactored through your lemma. Since that's a very specific statement, only needed for the next proof, I'd rather keep it in this file (and over the reals only) though.

Reply:
I encourage you to generalise + relocate `segment_inter_eq_endpoint_of_linearIndependent_sub` but I do not insist: I agree this is subjective.

Reply:
Moved and generalized.